include/MySQL_HostGroups_Manager.h

```diff
@@ -390,6 +390,80 @@ class MySQL_HostGroups_Manager {
 #else
 	rwlock_t rwlock;
 #endif
+
+	enum HGM_TABLES {
+		MYSQL_SERVERS = 0,
+		MYSQL_REPLICATION_HOSTGROUPS,
+		MYSQL_GROUP_REPLICATION_HOSTGROUPS,
+		MYSQL_GALERA_HOSTGROUPS,
+		MYSQL_AWS_AURORA_HOSTGROUPS,
+		MYSQL_HOSTGROUP_ATTRIBUTES,
+
+		__HGM_TABLES_SIZE
+	};
+
+	std::array<uint64_t, __HGM_TABLES_SIZE> table_resultset_checksum { 0 };
+
+	class HostGroup_Server_Mapping {
+	public:
+		enum Type {
+			WRITER = 0,
+			READER = 1,
+
+			__TYPE_SIZE
+		};
+
+		struct Node {
+			MySrvC* srv = NULL;
+			unsigned int reader_hostgroup_id = -1;
+			unsigned int writer_hostgroup_id = -1;
+			MySerStatus server_status = MYSQL_SERVER_STATUS_OFFLINE_HARD;
+		};
+
+		HostGroup_Server_Mapping(MySQL_HostGroups_Manager* hgm) : readonly_flag(1), myHGM(hgm) { }
+		~HostGroup_Server_Mapping() = default;
+
+		// Note: copy_if_not_exists, remove, clear method also makes changes to MyHostGroups
+		void copy_if_not_exists(Type dest_type, Type src_type);
```

Comment:
Could you please add Doxygen style documentation for this three methods? Specially for knowing which actions they might perform over `MySQL_Hostgroups_Manager`, and under which conditions, like server creation and server destruction.

Reply:
Done